tests/loader_envvar_tests.cpp

```diff
@@ -198,6 +198,21 @@ TEST(EnvVarICDOverrideSetup, XDG) {
     check_paths(env.debug_log, ManifestCategory::implicit_layer, HOME);
     check_paths(env.debug_log, ManifestCategory::explicit_layer, HOME);
 }
+// Check that a json file in the paths don't cause the loader to crash
+TEST(EnvVarICDOverrideSetup, XDGContainsJsonFile) {
+    // Set up a layer path that includes default and user-specified locations,
+    // so that the test app can find them.  Include some badly specified elements as well.
+    // Need to redirect the 'home' directory
+    set_env_var("XDG_CONFIG_DIRS", "bad_file.json");
```

Comment:
It seems a little strange to put a json path directly in the `XDG_CONFIG_DIRS` environment variable. I thought it was supposed to be a directory that contains json files.

Reply:
Well, this behavior is what triggered the crash in the loader. While XDG_*** are suppose to only have folders, it is possible there is a file in it. Thus, the loader tests should exercise that just in case.